src/attributes/classes.js

```diff
@@ -114,22 +114,31 @@ jQuery.fn.extend({
 		}
 
 		return this.each(function() {
-			var className, i, self, classNames;
+			var className, i, self, classNames, predicate;
 
 			if ( type === "string" ) {
 
 				// Toggle individual class names
 				i = 0;
 				self = jQuery( this );
+				predicate = jQuery.isFunction( stateVal ) ? stateVal.call( self, i, self) : null;
```

Comment:
To the point @Mr21 makes about `i = 0`, that does seem wrong. In other places where we allow functions, the function is called for each element.  Here we are only calling it once and saving the result.